rules/order.js

```diff
@@ -1,6 +1,181 @@
+
+    // Position, dimensions and boundaries
+    'position',
+    'top',
+    'right',
+    'bottom',
+    'left',
+
+    'width',
+    'min-width',
+    'max-width',
+    'height',
+    'min-height',
+    'max-height',
+
+    'z-index',
+    'overflow',
+    'overflow-x',
+    'overflow-y',
+
+    ...trbl('margin'),
+    ...trbl('padding'),
+
+    'transform',
+    'zoom',
+
+    // Display
+    'display',
+
+    // Flexbox only
+    'flex',
+    'flex-basis',
+    'flex-grow',
+    'flex-shrink',
+    'flex-flow',
+    'flex-wrap',
+    'flex-direction',
+    'justify-content',
+    'align-self',
+    'align-items',
+    'align-content',
+    'order',
+
+    // Grid Layout
+    'grid',
+    'grid-area',
+    'grid-auto-columns',
+    'grid-auto-flow',
+    'grid-auto-rows',
+    'grid-gap',
+    'grid-column',
+    'grid-column-end',
+    'grid-column-gap',
+    'grid-column-start',
+    'grid-row',
+    'grid-row-end',
+    'grid-row-gap',
+    'grid-row-start',
+    'grid-template',
+    'grid-template-areas',
+    'grid-template-columns',
+    'grid-template-rows',
+
+    'clear',
+    'float',
+    'visibility',
+    'vertical-align',
+    'direction',
+
+    // Table related stuff only
+    'table-layout',
+    'empty-cells',
+
+    // Asthethic
+    'list-style',
+    ...trbl('border'),
+    'outline',
+
+    'opacity',
+    'object-fit',
+    'object-position',
+    'clip-path',
+    'background',
+
+    'border-radius',
+    'box-shadow',
+    'filter',
+
+    // Text
+    'color',
+
+    'font',
+    'font-family',
+    'font-variant',
+    'font-size',
+    'font-size-adjust',
+    'font-kerning',
+    'font-weight',
+    'font-stretch',
+    'font-style',
+    'font-smoothing',
+
+    'text-decoration',
+    'text-underline-position',
+    'text-transform',
+    'text-shadow',
+    'text-overflow',
+    'text-align',
+    'text-justify',
+    'text-indent',
+
+    'line-height',
+    'tab-size',
+    'white-space',
+    'word-spacing',
+    'word-wrap',
+    'overflow-wrap',
+
+    // Paragraphs and wrapping
+    'word-break',
+    'hyphens',
+    'orphans',
+    'widows',
+
+    // Cursor
+    'pointer-events',
+    'user-select',
+    'cursor',
+    'resize',
+
+    // Animations & transitions
+    'animation',
+    'animation-name',
+    'animation-play-state',
+    'animation-direction',
+    'animation-fill-mode',
+    'animation-duration',
+    'animation-delay',
+    'animation-iteration-count',
+    'animation-timing-function',
+
+    'transition',
+    'transition-property',
+    'transition-duration',
+    'transition-delay',
+    'transition-timing-function',
+];
+
+/**
+ * Build top right bottom left variations plus shorthand version.
+ *
+ * @example ['padding', 'padding-top', 'padding-right', 'padding-bottom', 'padding-left']
+ *
+ * @param {String} prefix - Shorthand version ex. padding.
+ * @returns {String[]} - Array with all the variation.
+ */
+function trbl(prefix) {
+    const rules = [];
+
+    if (prefix) {
```

Comment:
This could be a ternary condition, but if you agree to use https://github.com/gilmoreorless/css-shorthand-properties, we will no longer need this function.